### PR TITLE
Patch unstructured bool

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
we patch the unstructured bool behavior so users can filter on the string bool field `{"string_bool": "true"}`

* **What is the current behavior?** (You can also link to an open issue here)
we can't filter the string_bool field(str)
bool field is not returned in the search results

* **What is the new behavior (if this is a feature change)?**
users can filter on the string bool field `{"string_bool": "true"}`
bool field will be returned in the search results

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
no

* **Related Python client changes** (link commit/PR here)
no

* **Related documentation changes** (link commit/PR here)
no

* **Other information**:
no

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: tests/tensor_search/integ_tests/test_search_unstructured.py [hunk 1]
```diff
@@ -519,6 +536,51 @@ def test_filtering(self):
                 if expected_id:
                     self.assertEqual(expected_id, res["hits"][0]["_id"])
 
+    def test_filtering_fake_and_real_boolean_fields(self):
```
Comment (farshidz) [inline tests/tensor_search/integ_tests/test_search_unstructured.py]: I find the fake and real boolean terminology confusing. How about `test_filter_string_boolean_fields`?

File: src/marqo/core/unstructured_vespa_index/unstructured_vespa_schema.py [hunk 2]
```diff
@@ -219,6 +219,7 @@ def _generate_unstructured_schema(cls, marqo_index: UnstructuredMarqoIndex) -> s
                     summary {cls._LONGS_STRINGS_FIELDS} type map<string, string> {{}}
                     summary {cls._SHORT_STRINGS_FIELDS} type map<string, string> {{}}
                     summary {cls._STRING_ARRAY} type array<string> {{}}
+                    summary {cls._BOOL_FIELDS} type map<string, byte> {{}}
```
Comment (farshidz) [inline src/marqo/core/unstructured_vespa_index/unstructured_vespa_schema.py:224]: Looks like tests have been missing this bug. We need to add tests to cover this now

**Review by farshidz — CHANGES_REQUESTED** [review]

File: tests/tensor_search/integ_tests/test_search_unstructured.py (hunk 1 above)
Comment (wanliAlex) [reply tests/tensor_search/integ_tests/test_search_unstructured.py]: done

File: src/marqo/core/unstructured_vespa_index/unstructured_vespa_schema.py (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/core/unstructured_vespa_index/unstructured_vespa_schema.py:224]: resolved

File: tests/tensor_search/integ_tests/test_search_unstructured.py [hunk 3]
```diff
@@ -519,6 +536,51 @@ def test_filtering(self):
...
+                        f"search_method = {search_method}, filter_string={filter_string}, "
+                        f"expected_hits={expected_hits}, expected_id={expected_id}")):
+                    res = tensor_search.search(
+                        index_name=self.default_text_index, config=self.config, text="search me",
+                        search_method=search_method, filter=filter_string
+                    )
+                    self.assertEqual(expected_hits, len(res["hits"]))
+                    if expected_id:
+                        self.assertEqual(expected_id, res["hits"][0]["_id"])
+                        expected_document = documents[0] if expected_id == "1" else \
```
Comment (farshidz) [inline tests/tensor_search/integ_tests/test_search_unstructured.py:580]: Testing the returned doc has all fields is a separate test case

**Review by farshidz — CHANGES_REQUESTED** [review]
As discussed please check structured tests and consider doing a full comparison of schemas rather than individual asserts

**Review by farshidz — DISMISSED** [review]

**Review by farshidz — APPROVED** [review]

**Pull request merged**